### PR TITLE
Move methods for metric calculations from Evaluator class

### PR DESCRIPTION
**Pull request by arangatang**
*Issue #, if available:*
#1135 & #1225 

*Description of changes:*
This PR aims to cleanup the Evaluator class and changing the way that metrics are being handled and calculated. 
This will be done in several steps as detailed below.

- [x] Move methods for calculating metrics outside of the Evaluator class
- [ ] Remove masking of denominators close to zero for MAPE, MASE, sMAPE, MSIS
- [ ] Write unit tests for all metrics (In progress)
- [ ] Write docstrings for metrics

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/gluonts/evaluation/_base.py [hunk 1]
```diff
@@ -31,9 +31,22 @@
 import numpy as np
 import pandas as pd
 
+from gluonts.evaluation.metrics import (
```
Comment (jaheba) [inline src/gluonts/evaluation/_base.py:34]: I think relative imports would make more sense here. So `from .metrics import ...`

**Review by jaheba — COMMENTED** [review]
Looks good so far. I think we can merge this in a first step and then do the other steps later.

**Review by lostella — COMMENTED** [review]
Looks good, I agree with proceeding with multiple PRs. @arangatang could you make sure the PR title & commit message reflects the change before merging? So it's easier when we do the next release

File: src/gluonts/evaluation/_base.py (hunk 1 above)
Comment (arangatang) [reply src/gluonts/evaluation/_base.py:34]: Sure thing, changed it!

**Review by lostella — APPROVED** [review]

**Pull request merged**